scene/main/window.cpp

```diff
@@ -649,12 +651,31 @@ void Window::_event_callback(DisplayServer::WindowEvent p_event) {
 		case DisplayServer::WINDOW_EVENT_FOCUS_IN: {
 			focused = true;
 			_propagate_window_notification(this, NOTIFICATION_WM_WINDOW_FOCUS_IN);
+			if (bool(GLOBAL_GET("audio/general/mute_on_focus_loss"))) {
+				// Unmute the Master bus.
+				print_verbose("Window: Unmuting audio on focus restore.");
+				AudioServer::get_singleton()->set_bus_mute(0, false);
+				// mute_on_focus_loss_is_tweening = true;
+				// Ref<Tween> tween = get_tree()->create_tween();
+				// AudioServer::get_singleton()->AudioServer::set_bus_mute(0, false);
+				// tween->tween_method(callable_mp(this, &Window::_smooth_audio_mute_callback), -80, mute_on_focus_loss_previous_db, 0.5);
```

Comment:
I've tried to smoothly fade the transition. While this code works, I couldn't figure out a way to have the tweens not "fight" each other when both are triggered at the same time (due to popups creating "focus in" notifications when the main window gets a "focus out").